src/runtime/GlobalObjectBuiltinAtomics.cpp

```diff
@@ -197,36 +194,61 @@ static Value atomicReadModifyWrite(ExecutionState& state, Value typedArray, Valu
         v = Value(value.toInteger(state));
     }
 
-    size_t elementSize = TA->elementSize();
-    size_t offset = TA->byteOffset();
-    size_t indexedPosition = (i * elementSize) + offset;
-
     return getModifySetValueInBuffer(state, buffer, indexedPosition, type, v, op);
 }
 
 static Value builtinAtomicsAdd(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
 {
-    return atomicReadModifyWrite(state, argv[0], argv[1], argv[2], AtomicOps::ADD);
+    return atomicReadModifyWrite(state, argv[0], argv[1], argv[2], AtomicBinaryOps::ADD);
 }
 
 static Value builtinAtomicsAnd(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
 {
-    return atomicReadModifyWrite(state, argv[0], argv[1], argv[2], AtomicOps::AND);
+    return atomicReadModifyWrite(state, argv[0], argv[1], argv[2], AtomicBinaryOps::AND);
+}
+
+static Value builtinAtomicsLoad(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    ArrayBufferObject* buffer = validateIntegerTypedArray(state, argv[0]);
+    TypedArrayObject* TA = argv[0].asObject()->asTypedArrayObject();
+    size_t indexedPosition = validateAtomicAccess(state, TA, argv[1]);
+    TypedArrayType type = TA->typedArrayType();
+
+    return buffer->getValueFromBuffer(state, indexedPosition, type);
 }
 
 static Value builtinAtomicsOr(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
 {
-    return atomicReadModifyWrite(state, argv[0], argv[1], argv[2], AtomicOps::OR);
+    return atomicReadModifyWrite(state, argv[0], argv[1], argv[2], AtomicBinaryOps::OR);
```

Comment:
How about check ```argc```? There are many similar things.

Reply:
Good point!
In this case, we don't need to check `argc` because this `builtinAtomicsOr` function is registered with length 3 later.
It means that `builtinAtomicsOr` function should be invoked with 3 arguments and these arguments are guaranteed even if  there are less than 3 arguments actually (lacking arguments will be initialized as `Undefined` value).

Reply:
Thanks for the detailed answer!